### ISSUE TITLE
 Relationship of the number of parameters and the resolution

### ISSUE THREAD

**Issue by walter2482** [ultralytics/ultralytics#1514]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Greetings, I hope you are very well, in advance I appreciate your contribution with your algorithm.

My question is related to the fact that when I work with a model, for example, YOLOv8m, it designates 25,843,813 parameters, both for 640 and 1280 resolution, what is this behavior due to in terms of neural architecture?

**Comment by N-Friederich** [comment]
Currently, YOLOv8 only provides pre-trained weights with 640^2 resolution.
An answer to your question depends on very many properties of your dataset you are training with. But in general: 
If the images in the original have a high resolution (like many COCO images) then it makes sense to use a higher resolution. Especially if there are small objects, which are then easier to recognize for the network.
Depending on the dataset, transfer learning can also lead to poorer results if the solution deviates too much (keyword: aspect ratio).

But these and other points depend on your data set. I think to give you a better answer, there needs more detail on your data set :)

**Comment by walter2482** [comment]
As for the data set, I'm investigating about the construction:
This are my dataset
![class_balance](https://user-images.githubusercontent.com/93685420/226463568-c53937ad-5df8-4962-9a3d-75ea060a4064.PNG)
The resolution of scale is this, i use scaling with black edges maintend proportion of images
![status_images](https://user-images.githubusercontent.com/93685420/226463656-3ef1e5ab-89e4-446b-9196-94f824d68f32.PNG)

My cmd for traing is this

yolo mode=train model=yolov8x.pt data=/notebooks/dataset/color_dataset/data.yaml  optimizer=SGD lr0=0,004375 lrf=0,004375 epochs=150 imgsz=1280 batch=32 save_txt=True seed=100 conf=0.001 iou=0.7 plots=True save_conf=True 

This are my result, please people dont steal me this xD
![models_train](https://user-images.githubusercontent.com/93685420/226463944-02857640-d592-41b3-91ee-d8ba6cf4bc7d.PNG)

**Comment by AyushExel** [comment]
The imgsz should not directly affect the model parameters but it'll increase the training time by a lot. That is because the computation performed by the sliding convolution kernels/filters, is directly proportional to the image size. 
There is a tradeoff though. You might get some accuracy boost on using higher resolution but in our experimentation that slight boost wasn't worth more than 2x training time.

**Comment by N-Friederich** [comment]
> As for the data set, I'm investigating about the construction: This are my dataset ![class_balance](https://user-images.githubusercontent.com/93685420/226463568-c53937ad-5df8-4962-9a3d-75ea060a4064.PNG) The resolution of scale is this, i use scaling with black edges maintend proportion of images ![status_images](https://user-images.githubusercontent.com/93685420/226463656-3ef1e5ab-89e4-446b-9196-94f824d68f32.PNG)
> 
> My cmd for traing is this
> 
> yolo mode=train model=yolov8m.pt data=/notebooks/dataset/color_dataset/data.yaml optimizer=SGD lr0=0,004375 lrf=0,004375 epochs=150 imgsz=1280 batch=32 save_txt=True seed=100 conf=0.001 iou=0.7 plots=True save_conf=True
> 
> This are my result, please people dont steal me this xD ![models_train](https://user-images.githubusercontent.com/93685420/226463944-02857640-d592-41b3-91ee-d8ba6cf4bc7d.PNG)

@walter2482 Ok, so the data in the original has a higher resolution than 640^2. My advice: do some trainings between 640 and 1280 (e.g. 3-4) and observe if there is a trend. As @AyushExel  said, in the end it always depends on your task context. If the computation time is less important, then it might be worth using a higher resolution. But with your amount of data, a model with more parameters might be worthwhile. In the end, there are many possible parameters that can lead to success, the resolution is only one.

**Comment by walter2482** [comment]
Once again thank you very much for your answers, at this time I will train a model with a greater number of parameters. I am very happy for your answers @AyushExel @N-Friederich

**Comment by Yee0717** [comment]
> 再次非常感谢您的回答，此时我将训练一个具有更多参数的模型。我很高兴你的回答

Hello, I am also doing relevant experiments. When improving the resolution of the input image, the parameters of the network will inevitably increase, but I do not know how to calculate this parameter. May I ask if you have solved it?

**Comment by glenn-jocher** [comment]
@Yee0717 当水平或垂直分辨率提高时，神经网络所需的参数数通常会增加。 在YOLOv8的情况下，这主要是由于卷积层的滤波器尺寸扩大。 
每个卷积层都有公式：total_output_channels x (filter_height x filter_width x input_channels + 1), 其中+1是偏移项。因此，总参数数量随着每个卷积层尺寸的增加而增加。 
您可以计算各层的参数量并将其相加，以获得总模型参数量。

**Comment by Yee0717** [comment]
> 

No words can express my thanks to your great kindness. Thank you so much.

**Comment by pderrenger** [comment]
@Yee0717 you're very welcome! If you have any more questions or need further assistance, feel free to ask. Good luck with your experiments!
